python_modules/libraries/dagster-dg/dagster_dg/cli/plus/deploy.py

```diff
@@ -163,9 +172,30 @@ def deploy_group(
         statedir,
         bool(use_editable_dagster),
         python_version,
+        location_names,
     )
 
-    finish_deploy_session(dg_context, statedir)
+    finish_deploy_session(dg_context, statedir, location_names)
+
+
+def _validate_location_names(
+    dg_context: DgContext, location_names: list[str], cli_config: DgRawCliConfig
+):
+    if not location_names:
+        return
+
+    if dg_context.is_project:
+        existing_location_names = {dg_context.code_location_name}
+    else:
+        existing_location_names = {
+            dg_context.for_project_environment(project_spec.path, cli_config).code_location_name
```

Comment:
we ought to just have a `all_project_contexts` util or something, since I end up doing this in a few places too - can handle in follow up (I can do in hybrid scaffold PR)